packages/common/src/location/platform_location.ts

```diff
@@ -33,11 +33,7 @@ import {DOCUMENT} from '../dom_tokens';
  *
  * @publicApi
  */
-@Injectable({
-  providedIn: 'platform',
-  // See #23917
-  useFactory: useBrowserPlatformLocation
-})
+@Injectable({providedIn: 'platform', useFactory: () => new BrowserPlatformLocation()})
```

Comment:
```suggestion
@Injectable({providedIn: 'platform', useFactory: () => inject(BrowserPlatformLocation)})
```
Should still avoid using `new`